install/kubernetes/helm/subcharts/kiali/templates/deployment.yaml

```diff
@@ -48,11 +48,13 @@ spec:
             secretKeyRef:
               name: {{ .Values.dashboard.secretName }}
               key: {{ .Values.dashboard.usernameKey }}
+              optional: true
```

Comment:
I RTD at https://kubernetes.io/docs/concepts/configuration/secret/ - however, it is a little slim on details of the symantics of the optional secretKeyRef.  Will this result in a mounting if the secrets if they are dynamically created?

The symantics I believe we agreed upon were to use created secrets if they were present, if not present to start up but not provide any access to the system.  Once the secreets are registered, the secrets would then be mounted into the deployment and be usable.

Does this PR do that?

Reply:
For background on why this "optional=true" was added to the secret-backed env var definitions, see https://github.com/istio/istio/issues/11244 and [KIALI-2303](https://issues.jboss.org/browse/KIALI-2303)

What happens is: IF the secret does NOT exist, the pod will still start, however the environment variables based on the secret will NOT be defined. Kiali will detect this condition and will not provide any access to the system - instead, the user will get [an error message in the login screen](https://issues.jboss.org/secure/attachment/12446051/Screenshot%20from%202019-01-25%2015-44-47.png) indicating the problem. Once the secret is created, the pod is required to be restarted (e.g. `kubectl delete pod --selector=app=kiali` will cause the pod to be scaled up back to 1 and access will now be granted assuming the secret exists and the user provides the valid credentials at login time). I do not know of a way to get kubernetes to auto-restart a pod if it detects a secret that was missing has been created during the lifetime of a pod.

Reply:
@jmazzitelli sounds like a good interrmediate solution.  I also don't know of any way to do this with Kubuernetes either using manifests.  We have used a solution like the following to solve this problem though with go code:

Watch for secret registration with the correct name in the correct namespace - if it is present, then use it.  Check out:
https://github.com/istio/istio/tree/master/pkg/kube/secretcontroller

For an example.

Cheers
-steve